File: src/design_patterns/behavioral/command/c#/command.cs
abstract class Command
{
    public abstract void Execute();
}

class OpenFileCommand : Command
{
    private string filename;

    public OpenFileCommand(string filename)
    {
        this.filename = filename;
    }

    public override void Execute()
    {
        Console.WriteLine($"Opening file {filename}");
    }
}

class CloseFileCommand : Command
{
    private string filename;

    public CloseFileCommand(string filename)
    {
        this.filename = filename;
    }

    public override void Execute()
    {
        Console.WriteLine($"Closing file {filename}");
    }
}

class FileManager
{
    private List<Command> commands = new List<Command>();

    public void AddCommand(Command command)
    {
        commands.Add(command);
    }

    public void ExecuteCommands()
    {
        foreach (Command command in commands)
        {
            command.Execute();
        }
    }
}

FileManager fileManager = new FileManager();
fileManager.AddCommand(new OpenFileCommand("file1.txt"));
fileManager.AddCommand(new CloseFileCommand("file1.txt"));
fileManager.AddCommand(new OpenFileCommand("file2.txt"));
fileManager.ExecuteCommands();

File: src/design_patterns/behavioral/command/python/command.py
class Command:
    def execute(self):
        pass


class OpenFileCommand(Command):
    def __init__(self, filename):
        self.filename = filename

    def execute(self):
        print(f"Opening file {self.filename}")


class CloseFileCommand(Command):
    def __init__(self, filename):
        self.filename = filename

    def execute(self):
        print(f"Closing file {self.filename}")


class FileManager:
    def __init__(self):
        self.commands = []

    def add_command(self, command):
        self.commands.append(command)

    def execute_commands(self):
        for command in self.commands:
            command.execute()


file_manager = FileManager()
file_manager.add_command(OpenFileCommand("file1.txt"))
file_manager.add_command(CloseFileCommand("file1.txt"))
file_manager.add_command(OpenFileCommand("file2.txt"))
file_manager.execute_commands()
